### DISCUSSION TITLE
terser ES6 config

### DISCUSSION THREAD

**Discussion by PuiuCS** [holger1411/understrap#1689] [Q&A] [answered]
Hello,
I'm using the newest child theme and i'm trying to include some custom ES6 js code.  

I've tried changing the value of "ecma" in the terser.config.json file from 5 to 6, 2015, es6 etc. I even added "module" : true , but the JS doesn't seem to be to be properly minified. Is there something else i should do? If i write normal vanilla js it works.

Here's an example of a simple ES6 JS code:

(function(){
    Home = {
        init: () =>{
            console.log('Hello');
            Home.homeStart();
        },
        homeStart:() =>{
            console.log('homeStart');
            document.addEventListener("scroll", () =>{
                Home.scrollHomeFunction();
           });
        },
        scrollHomeFunction: () =>{
            console.log(window.scrollY);
        }
    };
    Home.init();

 })();

**Comment by IanDelMar** [comment, marked as answer]
Try this
```diff
(function(){
-	Home = {
+	const Home = {
		init: () =>{
			console.log('Hello');
			Home.homeStart();
		},
		homeStart:() =>{
			console.log('homeStart');
			document.addEventListener("scroll", () =>{
				Home.scrollHomeFunction();
			});
		},
		scrollHomeFunction: () =>{
			console.log(window.scrollY);
		}
	};
	Home.init();

})();
```

**Reply by PuiuCS** [reply]
Thank you. It finally worked. It was a mix of errors on my part. The footer wasn't loading properly.
